### PR TITLE
EVG-16120 Sticky column headers and refactor commits page

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-16120](https://jira.mongodb.org/browse/EVG-16120)

### Description 
Refactored the commits page to have a more semantically correct html structure. Which now allows us to use the `position: sticky` css attribute.  

This also updates some sizes in preparation for EVG-16152
There is also a refactor of some storybook stories and some components to make it easier to render the project health view in a storybook story.
### Screenshots

https://user-images.githubusercontent.com/4605522/150557960-a1a713b0-19f3-41fe-9747-0d88048b3338.mov

### CODE REVIEW

File: src/pages/commits/CommitsWrapper.tsx [hunk 1]
```diff
@@ -28,91 +36,130 @@ export const CommitsWrapper: React.FC<Props> = ({
...
+const getCommitWidth = (commit: Commit) => {
+  const { version, rolledUpVersions } = commit;
+  if (version) {
+    return 200;
+  }
+  if (rolledUpVersions) {
+    return 64;
+  }
+  throw new Error("Commit type not found");
+};
```
Comment (minnakt) [inline src/pages/commits/CommitsWrapper.tsx:127]: It seems like `ChartContainer` in `CommitChart.tsx` is inheriting this `200px` width and it's causing a bug where the`Tooltip` is pretty far away from the actual bars in the graph (differs from current Spruce behavior)


Screen Shot 2022-01-21 at 4 51 47 PM

I don't know if this is the best solution but adding `width: fit-content` to the `ChartContainer` does fix this issue!

Screen Shot 2022-01-21 at 5 00 32 PM

File: src/pages/commits/CommitsWrapper.tsx [hunk 2]
```diff
@@ -28,91 +36,130 @@ export const CommitsWrapper: React.FC<Props> = ({
...
+    }
+    return undefined;
+  }, [versions]);
+
+  const maxGroupedTaskStats = useMemo(() => {
+    if (versionToGroupedTaskStatsMap) {
+      return findMaxGroupedTaskStats(versionToGroupedTaskStatsMap);
+    }
+    return undefined;
+  }, [versionToGroupedTaskStatsMap]);
```
Comment (minnakt) [inline src/pages/commits/CommitsWrapper.tsx:53]: are the `return undefined`s needed here? I removed them and observed the same behavior with or without, but I might have overlooked something

**Review by minnakt — COMMENTED** [review]
I don’t have much to say but it looks great! I already thought it looked good before, but now it looks even better because the scrolling became really seamless. 🚀 The refactor makes a lot of sense considering the sticky functionality. 🙂

File: src/pages/commits/CommitsWrapper.tsx [hunk 3]
```diff
@@ -28,91 +36,130 @@ export const CommitsWrapper: React.FC<Props> = ({
...
 
-export const ProjectHealthWrapper = styled.div`
-  display: flex;
-  flex-direction: column;
-  justify-content: flex-start;
-  align-items: flex-start;
-  height: 100%;
-  width: 100%;
-  position: relative;
+const CommitWrapper = styled.div<{ width: number }>`
```
Comment (sophstad) [inline src/pages/commits/CommitsWrapper.tsx:155]: I like this syntax, much cleaner than including typing in the arrow function 👌

File: src/pages/commits/ActiveCommits/ProjectHealth.stories.tsx [hunk 4]
```diff
@@ -1,77 +1,102 @@
...
+      hasTaskFilter={boolean("hasTaskFilter", false)}
+      hasFilters={boolean("hasFilters", false)}
+      onChangeChartType={setChartType}
+    />
+  );
+};
+
+type Commits = MainlineCommitsQuery["mainlineCommits"]["versions"];
+type Commit = Commits[0];
+type Version = Commit["version"];
```
Comment (sophstad) [inline src/pages/commits/ActiveCommits/ProjectHealth.stories.tsx:38]: Should these be imported from `types/commits.ts`?

File: src/types/commits.ts [hunk 5]
```diff
@@ -24,3 +26,8 @@ export enum CommitRequesterTypes {
   TriggerRequester = "trigger_request",
   GitTagRequester = "git_tag_request",
 }
+
+export type Commits = MainlineCommitsQuery["mainlineCommits"]["versions"];
+export type Commit = Commits[0];
```
Comment (sophstad) [inline src/types/commits.ts:31]: [nit] You could probably use the [Unpacked](https://github.com/evergreen-ci/spruce/blob/main/src/types/utils.ts#L2) helper instead

**Review by sophstad — APPROVED** [review]
LGTM once Minna's note about the tooltip position is addressed! I also think it could look nice to apply a small shadow to the bottom of the commit row, kinda like GitHub does:
image
But up to you if you want to implement that!

**Comment by khelif96** [comment]
>I also think it could look nice to apply a small shadow to the bottom of the commit row, kinda like GitHub does: image But up to you if you want to implement that!

I looked into this unfortunately there isn't a clean way to toggle the box shadow based off of if the element is scrolled or not. One of the solutions i found would be to attach an event handler to the element and measure if its scrolled but at that point it becomes the solution i have outlined in the other pr. If design asks for it i can take a look at it. I do think it would be really neat if it did have the box shadow tho but it might not be worth the added complexity.

**Review by minnakt — APPROVED** [review]

**Pull request merged**